enumerator/usb_darwin.go

```diff
@@ -18,6 +18,26 @@ import (
 	"unsafe"
 )
 
+type io_iterator_t struct {
```

Comment:
I think we could define the go-lang type directly to the C type (without wrapping them into a struct)

```go
type io_iterator_t C.io_iterator_t
```

this will allow to change the boxing/unboxing into a cast, for example:

```go
&io_registry_entry_t{ioregistryentry: service.ioobject})
```

may be rewritten as:

```go
io_registry_entry_t(service)
```

or to transform back to the original C type:

```go
C.io_registry_entry_t(service)
```